connection/connection.go

```diff
@@ -0,0 +1,71 @@
+package connection
```

Comment:
I think we need the usual copyright and license boiler plate code at the beginning of this file.